cmd/prometheus/main.go

```diff
@@ -282,104 +277,193 @@ func main() {
 		notifier,
 	}
 
-	// Wait for reload or termination signals. Start the handler for SIGHUP as
-	// early as possible, but ignore it until we are ready to handle reloading
-	// our config.
-	go func() {
-		<-hupReady
-		for {
-			select {
-			case <-hup:
-				if err := reloadConfig(cfg.configFile, logger, reloadables...); err != nil {
-					level.Error(logger).Log("msg", "Error reloading config", "err", err)
-				}
-			case rc := <-webHandler.Reload():
-				if err := reloadConfig(cfg.configFile, logger, reloadables...); err != nil {
-					level.Error(logger).Log("msg", "Error reloading config", "err", err)
-					rc <- err
-				} else {
-					rc <- nil
-				}
-			}
-		}
-	}()
+	prometheus.MustRegister(configSuccess)
+	prometheus.MustRegister(configSuccessTime)
 
 	// Start all components while we wait for TSDB to open but only load
 	// initial config and mark ourselves as ready after it completed.
 	dbOpen := make(chan struct{})
+	// Wait for reload or termination signals. Start the handler for SIGHUP as
```

Comment:
This channel/handler doesn't seem to be related to termination signals, but only to reloads.